packages/field-label/src/FieldLabel.ts

```diff
@@ -79,13 +87,24 @@ export class FieldLabel extends SizedMixin(SpectrumElement) {
         }
     }
 
-    private async manageFor(): Promise<void> {
-        if (!this.for) {
-            return;
-        }
-        const parent = this.getRootNode() as HTMLElement;
-        const target = parent.querySelector(`#${this.for}`) as Focusable;
+    private resolvedElement = new ElementResolutionController(this);
+
+    private async manageTarget(): Promise<void> {
+        const target = this.resolvedElement.element as Focusable;
         if (!target) {
+            if (this.target) {
```

Comment:
Should the previous target be cleaned regardless? Check for identity change instead?